src/Orleans/IDs/UniqueKey.cs

```diff
@@ -114,7 +114,7 @@ private static UniqueKey NewKey(ulong n0, ulong n1, Category category, long type
             // in the string representation of a key, we grab the least significant half of n1.
             // therefore, if n0 is non-zero and n1 is 0, then the string representation will always be
             // 0x0 and not useful for identification of the grain.
-            if (n1 == 0 && n1 != 0)
+            if (n1 == 0 && n0 != 0)
                 throw new ArgumentException("n0 cannot be zero unless n1 is non-zero.", "n0");
```

Comment:
If I read the exception message correctly, then the comparison should be inverted (essentially `n0 == 0 && n1 != 0`). Either that, or the exception message also needs fixing.